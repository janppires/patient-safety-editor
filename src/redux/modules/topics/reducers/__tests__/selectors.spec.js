import * as topicsSelector from '../index';

describe('topics selectors', () => {
  test('get topics array from state', () => {
    const state = prepareStateWithTopics();
    const topics = topicsSelector.getTopics(state);
    expect(topics).toMatchSnapshot();
  })

  test('get undefined topic when no topic is selected', () => {
    const state = prepareStateWithTopics();
    const topic = topicsSelector.getSelectedTopic(state);
    expect(topic).toMatchSnapshot();
  })

  test('get selected topic', () => {
    const state = prepareStateWithTopicsAndSelectedTopic();
    const topic = topicsSelector.getSelectedTopic(state);
    expect(topic).toMatchSnapshot();
  })

  const prepareStateWithTopics = () => ({
    topics: {
      list: [1, 2],
      byId: {
        1: { name: 'random' }, 
        2: { name: 'random 2' }
      },
      selected: null
    }
  })

  const prepareStateWithTopicsAndSelectedTopic = () => {
    const state = prepareStateWithTopics();
    state.topics.selected = 2;
    return state;
  }
})