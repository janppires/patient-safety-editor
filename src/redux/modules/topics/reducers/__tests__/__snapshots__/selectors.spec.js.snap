// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`topics selectors get selected topic 1`] = `
Object {
  "name": "random 2",
}
`;

exports[`topics selectors get topics array from state 1`] = `
Array [
  Object {
    "name": "random",
  },
  Object {
    "name": "random 2",
  },
]
`;

exports[`topics selectors get undefined topic when no topic is selected 1`] = `undefined`;
